hc/accounts/tests/test_login.py

```diff
@@ -16,22 +17,42 @@ def test_it_sends_link(self):
 
         form = {"email": "alice@example.org"}
 
-        r = self.client.post("/accounts/login/", form)
-        assert r.status_code == 302
+        create_user = self.client.post("/accounts/login/", form)
+        assert create_user.status_code == 302
 
         ### Assert that a user was created
-
+        assert Profile.objects.select_related("user").get(user__email=form["email"])  # Go to database and retrieve user
+                                                                                      # hc/accounts/backends.py:21
         # And email sent
         self.assertEqual(len(mail.outbox), 1)
         self.assertEqual(mail.outbox[0].subject, 'Log in to healthchecks.io')
         ### Assert contents of the email body
-
+        self.assertIn("please open the link below", str(mail.outbox[0].body))
         ### Assert that check is associated with the new user
+        user_alice = User.objects.get(email="alice@example.org")
+        check.refresh_from_db()
+        self.assertEqual(check.user, user_alice)
 
     def test_it_pops_bad_link_from_session(self):
         self.client.session["bad_link"] = True
         self.client.get("/accounts/login/")
         assert "bad_link" not in self.client.session
 
         ### Any other tests?
+    def test_invalid_email_during_login_should_not_be_accepted(self):
+        """
+        app should not accept an improper email when logging in
+        :return: redirect back to the login page
+        """
+        check = Check()
+        check.save()
+
+        session = self.client.session
+        session["welcome_code"] = str(check.code)
+        session.save()
+
+        form = {"email": "I_AM_PRETENDING_TO_BE_AN_EMAIL"}
 
+        invalid_email_login = self.client.post("/accounts/login/", form)
+        self.assertEqual(invalid_email_login.status_code, 200)  # returns status code 200 instead of 301/302
```

Comment:
These comments look like they're for your understanding, so maybe remove them..or change the comments to match the application context

Reply:
Okay thanks

Reply:
@sir3n-sn you could also check if it redirects the user to log in again.